### PR TITLE
Add updated date to codelist version template

### LINKED ISSUE #2106: Show codelist creation date
From [this Slack thread](https://bennettoxford.slack.com/archives/C07AA91BZD4/p1727955808431179):

> It would be very useful if we could see the date a codelist was created. There is the sign off date, but that only applies to the original codelist, i'm more interested in the dates of different versions, to know when it was last updated.

Could be on either or both of the landing page index and in the codelist version view. Landing page index could allow sorting or filtering by dates?

We have [`CodelistVersion.created_at`](https://github.com/opensafely-core/opencodelists/blob/cd038a6670ba17ff68cfd283a669ad245b46d004/codelists/models.py#L221). Versions also have `tag` and `id`.

Some version identifiers seem to include a date, but not all. E.g., [this](https://www.opencodelists.org/codelist/opensafely/ace-inhibitor-medications/2020-05-19/) vs. [that](https://www.opencodelists.org/codelist/openprescribing/ace-inhibitors/2bb0d404/).

### PR DESCRIPTION
**Pull request by Jongmassey**
Fixes #2106 

The codelist versions are rendered in date creation order, but this is not always obvious to users.
Even then, we do not surface the creation of modification date anywhere.

For a user, trying to find a version of a codelist at a particular point in time, this change provides the neccesary information.

image

### CODE REVIEW

File: templates/codelists/version.html [hunk 1]
```diff
@@ -211,6 +211,7 @@ <h6>Versions</h6>
         {% elif version.is_draft %}
         <span class="badge badge-primary">Draft</span>
         {% endif %}
+        <span>(updated {{ version.updated_at | date:'Y-m-d' }})
```
Comment (StevenMaude) [inline templates/codelists/version.html:214]: Unless I'm missing something, maybe this needs the closing tag?

```suggestion
        <span>(updated {{ version.updated_at | date:'Y-m-d' }})</span>
```

**Review by StevenMaude — APPROVED** [review]
This mostly looks fine.

But in the example screenshot, both codelists have the same date; I can't obviously tell which version is newer. (I could have if they had different dates, granted.)

Is it worth including the update time too?

File: templates/codelists/version.html (hunk 1 above)
Comment (Jongmassey) [reply templates/codelists/version.html:214]: Oh wow, don't know how I missed that one!

**Comment by Jongmassey** [comment]
Here's it with the time:

image


I was trying to keep things as small and un-ugly as possible, but as you said there are genuine reasons to show the time so I think it's unavailable.

**Pull request merged**